apps/nextjs/src/middleware.ts

```diff
@@ -1,31 +1,68 @@
-import * as Sentry from "@sentry/nextjs";
-import { authMiddleware } from "middlewares/auth.middleware";
-import { addCspHeaders } from "middlewares/csp";
 import { NextMiddlewareResult } from "next/dist/server/web/types";
-import { NextMiddleware, NextResponse } from "next/server";
+import {
+  NextFetchEvent,
+  NextMiddleware,
+  NextRequest,
+  NextResponse,
+} from "next/server";
 
+import { logError } from "./lib/errors/server/middlewareErrorLogging";
 import { sentryCleanup } from "./lib/sentry/sentryCleanup";
+import { authMiddleware } from "./middlewares/auth.middleware";
+import { CspConfig, addCspHeaders } from "./middlewares/csp";
 
-function handleError(error: unknown, extra: Record<string, unknown>): Response {
-  if (error instanceof SyntaxError) {
-    console.error("Handled SyntaxError in nextMiddleware", error);
+export type ErrorWithPotentialCause = Error & {
+  cause?: unknown;
+};
+
+function determineErrorResponse(error: unknown): Response {
+  const errorWithCause = error as ErrorWithPotentialCause;
+
+  if (
+    error instanceof SyntaxError ||
+    errorWithCause.cause instanceof SyntaxError
+  ) {
     return NextResponse.json({ error: "Bad Request" }, { status: 400 });
   }
 
-  const wrappedError = new Error("Error in nextMiddleware", { cause: error });
-  Sentry.captureException(wrappedError, { extra });
-
   return NextResponse.json({ error: "Internal Server Error" }, { status: 500 });
 }
 
+const cspConfig: CspConfig = {
```

Comment:
Moving this to a config allows us to test the CSP generation without relying on ENV being set.